Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderBuildArgumentsManager.h

```diff
@@ -78,6 +78,11 @@ namespace AZ
             //! then it is expected that this function should be called twice, as follows:
             //!     PushArguments("Windows")
             //!     PushArguments("dx12")
+            //! The names come from the directory structure under the Platform/ folder:
+            //! - Platform/
+            //!   - Windows/  (Platform name)
+            //!     - dx12/   (RHI name)
+            //!     - vulkan/ (RHI name)
             //! @returns The resulting (combined, with - and +) set of arguments at the top of the stack.
             const AZ::RHI::ShaderBuildArguments& PushArguments(const AZStd::string& name);
```

Comment:
These functions would be more clear as PushArgumentScope, PopArgumentScope since you aren't actually adding new arguments, but just entering a scope that already exists.
Also, it would help to have a summary statement in the comment. Right now it just starts with "@param"

Reply:
Fixed